avm/res/db-for-my-sql/flexible-server/main.bicep

```diff
@@ -324,12 +330,11 @@ resource flexibleServer 'Microsoft.DBforMySQL/flexibleServers@2023-12-30' = {
           startMinute: maintenanceWindow.customWindow == 'Enabled' ? maintenanceWindow.startMinute : 0
         }
       : null
-    network: !empty(delegatedSubnetResourceId) && empty(firewallRules)
-      ? {
-          delegatedSubnetResourceId: delegatedSubnetResourceId
-          privateDnsZoneResourceId: privateDnsZoneResourceId
-        }
-      : null
+    network: {
+      delegatedSubnetResourceId: !empty(delegatedSubnetResourceId) ? delegatedSubnetResourceId : null
+      privateDnsZoneResourceId: !empty(privateDnsZoneResourceId) ? privateDnsZoneResourceId : null
```

Comment:
I recommend changing both parameters to be nullable (e.g, `param delegatedSubnetResourceId string?`). 
If done, you can simplify these two lines to
```bicep
delegatedSubnetResourceId: delegatedSubnetResourceId
privateDnsZoneResourceId: privateDnsZoneResourceId
```

without changing the user experience.

Reply:
The same comment would apply to e.g. `sourceServerResourceId` and similar other parameters

Reply:
Done and I think I caught the other examples of this.